chain/syncer.go

```diff
@@ -434,29 +395,14 @@ func (syncer *Syncer) HandleNewTipset(ctx context.Context, tsKey types.TipSetKey
 			return err
 		}
 		if i%500 == 0 {
-			logSyncer.Infof("processing block %d of %v for chain with head at %v", i, len(chain), tsKey.String())
+			logSyncer.Infof("processing block %d of %v for chain with head at %v", i, len(chain), ph.Head.String())
 		}
 		parent = ts
 	}
 	return nil
 }
 
-func (syncer *Syncer) exceedsFinalityLimit(newHead types.TipSet) (bool, error) {
-	if !newHead.Defined() {
-		return false, errors.New("cannon check finality limit on undefined tipset")
-	}
-	curHead, err := syncer.chainStore.GetTipSet(syncer.chainStore.GetHead())
-	if err != nil {
-		return false, err
-	}
-	curHeight, err := curHead.Height()
-	if err != nil {
-		return false, err
-	}
-	finalityHeight := types.NewBlockHeight(curHeight).Add(types.NewBlockHeight(uint64(FinalityLimit)))
-	newHeight, err := newHead.Height()
-	if err != nil {
-		return false, err
-	}
-	return types.NewBlockHeight(newHeight).GreaterThan(finalityHeight), nil
+func (syncer *Syncer) exceedsFinalityLimit(curHeight, newHeight types.Uint64) bool {
+	finalityHeight := types.NewBlockHeight(uint64(curHeight)).Add(types.NewBlockHeight(uint64(FinalityLimit)))
+	return types.NewBlockHeight(uint64(newHeight)).GreaterThan(finalityHeight)
```

Comment:
Just use `uint64` arithmetic. This doesn't need to be encoded on chain and 64 bits ought to be enough for anyone's chain length.